docs/libs/ml/quickstart.md

```diff
@@ -24,4 +24,198 @@ under the License.
 * This will be replaced by the TOC
 {:toc}
 
-Coming soon.
+## Introduction
+
+FlinkML is designed to make learning from your data a straight-forward process, abstracting away
+the complexities that usually come with having to deal with big data learning tasks. In this
+quick-start guide we will show just how easy it is to solve a simple supervised learning problem
+using FlinkML. But first some basics, feel free to skip the next few lines if you're already
+familiar with Machine Learning (ML)
+
+As defined by Murphy [cite ML-APP] ML deals with detecting patterns in data, and using those
+learned patterns to make predictions about the future. We can categorize most ML algorithms into
+two major categories: Supervised and Unsupervised Learning.
+
+* Supervised Learning deals with learning a function (mapping) from a set of inputs
+(predictors) to a set of outputs. The learning is done using a __training set__ of (input,
```

Comment:
It's more of a statistics terminology, see [synonyms](http://en.wikipedia.org/wiki/Dependent_and_independent_variables#Statistics_synonyms). In ML features is more common so I will change it to that.